CMakeLists.txt

```diff
@@ -15,11 +15,17 @@ option(LIBCOSIM_BUILD_PRIVATE_APIDOC "Build private API documentation (only used
 option(LIBCOSIM_STANDALONE_INSTALLATION "Whether to build for a standalone installation (Linux only; sets a relative RPATH)" OFF)
 option(LIBCOSIM_USING_CONAN "Whether Conan is used for package management" OFF)
 option(LIBCOSIM_WITH_PROXYFMU "Whether or not to build with proxy-fmu integration" OFF)
+option(LIBCOSIM_FMI_LOGGING "Enabling FMI logging during simulation" ON)
 
 # ==============================================================================
 # Global internal configuration
 # ==============================================================================
 
+# Configure logging in the fmi callback function
+if(LIBCOSIM_FMI_LOGGING)
+    add_compile_definitions(LIBCOSIM_FMI_LOGGING=1)
```

Comment:
I think the C++ preprocessor macro should be called `LIBCOSIM_NO_FMI_LOGGING` instead, so that logging is enabled by default. (Practically, this just means that the `#ifdef` tests should be replaced with `#ifndef`.)

Note that I think it's fine that the _CMake_ variable is called `LIBCOSIM_FMI_LOGGING`, since its default value is defined by the `option()` command. That's not the case for macros, they are undefined by default.

Reply:
They are updated.